Source Code/Ticket Management System/supportgiver_menu.c

```diff
@@ -25,11 +25,33 @@ bool filterByStatus(void* ticket, void* status)
 {
 	return wcscmp(ticket_getStatus(ticket), status) != 0;
 }
+bool filterByTier(void* ticket1, void* tier)
+{
+	return wcscmp(ticket_getTier(ticket1), ticket_getTier(tier)) != 0;
+}
+bool filterByType(void* ticket1, void* type)
+{
+	return wcscmp(ticket_getType(ticket1), ticket_getType(type)) != 0;
 
+}
 bool sortByTitle(void* ticket1, void* ticket2)
 {
 	return wcscmp(ticket_getTitle(ticket1), ticket_getTitle(ticket2)) > 0;
 }
+bool filterByStakeholders(void* ticket1, void* stakeholders)
+{
+	return wcscmp(ticket_getStakeholders(ticket1), ticket_getStakeholders(stakeholders))!= 0;
```

Comment:
`stakeholders` is not a ticket, so you can't call `ticket_getStakeholders` on it.